### ISSUE TITLE
Provide the ability to sync the cache to disk

### ISSUE THREAD

**Issue by zackehh** [zackehh/cachex#92]
It would be nice if Cachex could automatically back itself up to a local file on some interval, as configured by the user. This should be disabled by default, and it should error if the file to load does not exist (to avoid confusing people). There are cases where this would be nice; in case of servers which are paused on Heroku for example. 

I would think this would be controlled entirely via cache options, but perhaps there's a case to be made for interface functions you can use to flush/load manually (I'll think on this, immediate example would be inside an `exrm` shell).

The options would be something like this (don't take this as a promise, yet):

```
# defaults to some sync interval
Cachex.start(:cache, [ disk_sync: "/tmp/dump.dat" ])

# custom sync interval in ms
Cachex.start(:cache, [ disk_sync: { "/tmp/dump.dat", 2000 } ])
```

**Comment by osayame** [comment]
@zackehh Are you actively working on this? If not, I'd like to grab this one.

**Comment by zackehh** [comment]
@osayame I have some stuff stashed on it but ran out of time; do you need it for use?

If so I'll polish what I have ASAP.

**Comment by osayame** [comment]
@zackehh I would like to use it for an active project at work. I'd appreciate it.

**Comment by zackehh** [comment]
@osayame you bet, I'll get back on it as soon as possible!

**Comment by osayame** [comment]
@zackehh Alternatively, you could push up the branch, and I can pick up where you left off.

**Comment by zackehh** [comment]
Aloha @osayame!

I'll have this done shortly - I have some time to work on it today; it should make it into master within a couple of days (if not today).

One note; you'll have to use this from master directly - is that ok?

**Comment by zackehh** [comment]
@osayame I'm curious; do you want the scheduling with this, or would you just want a manual flush/load option?

I'm starting to veer towards manually being able to backup to disk, and being able to load from a backup at startup - that way a user can craft their own scheduling policy.

Thoughts?

**Comment by osayame** [comment]
@zackehh Manually scheduling makes sense, and I'm onboard with using master directly.

**Comment by zackehh** [comment]
@osayame perfect, will have it up by the end of the day (PST)

**Comment by zackehh** [comment]
There is a PR for this in #100. 

For future readers, this issue changed from an automated interval into manual dumping and loading for now. In future the sync interval might be added, but for now it seems more appropriate that the caller just runs a dump when appropriate,

**Comment by zackehh** [comment]
Hi @osayame, this is now tagged and released on Hex as `v2.1.0`; want to try it out and let me know how it goes?

Thanks!
